src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ConcurrencyLimiter.cs

```diff
@@ -27,13 +27,18 @@ public sealed class ConcurrencyLimiter : RateLimiter
         private static readonly ConcurrencyLease SuccessfulLease = new ConcurrencyLease(true, null, 0);
         private static readonly ConcurrencyLease FailedLease = new ConcurrencyLease(false, null, 0);
         private static readonly ConcurrencyLease QueueLimitLease = new ConcurrencyLease(false, null, 0, "Queue limit reached");
-        private static readonly double TickFrequency = (double)TimeSpan.TicksPerSecond / Stopwatch.Frequency;
 
         // Use the queue as the lock field so we don't need to allocate another object for a lock and have another field in the object
         private object Lock => _queue;
 
         /// <inheritdoc />
-        public override TimeSpan? IdleDuration => _idleSince is null ? null : new TimeSpan((long)((Stopwatch.GetTimestamp() - _idleSince) * TickFrequency));
+        public override TimeSpan? IdleDuration => _idleSince is null ? null :
+#if NET
+            Stopwatch.GetElapsedTime(_idleSince.Value)
+#else
+            RateLimiterHelper.GetElapsedTime(_idleSince.Value)
+#endif
```

Comment:
Is it worth ifdefing all these places rather than just always using the internal `RateLimiterHelper.GetElapsedTime`?

Reply:
Did a small benchmark, benchmark code:

```csharp
[SimpleJob]
[MemoryDiagnoser]
public class StopwatchElapsedTimeTest
{
    private static readonly long StartTime = System.Diagnostics.Stopwatch.GetTimestamp();
    
    [Benchmark(Baseline = true)]
    public TimeSpan StopwatchTest()
    {
        return Stopwatch.GetElapsedTime(StartTime);
    }
    
    [Benchmark]
    public TimeSpan WatchGetElapsedTimeTest()
    {
        return StopwatchHelper.WatchGetElapsedTime(StartTime);
    }
    
    [Benchmark]
    public TimeSpan WatchGetElapsedTime2Test()
    {
        return StopwatchHelper.WatchGetElapsedTime2(StartTime);
    }
    
    [Benchmark]
    public TimeSpan CustomGetElapsedTimeTest()
    {
        return StopwatchHelper.CustomGetElapsedTime(StartTime);
    }
    
    [Benchmark]
    public TimeSpan CustomGetElapsedTime2Test()
    {
        return StopwatchHelper.CustomGetElapsedTime2(StartTime);
    }
}

file static class StopwatchHelper
{
    private static readonly double TickFrequency = (double)TimeSpan.TicksPerSecond / Stopwatch.Frequency;

    public static TimeSpan CustomGetElapsedTime(long startTimestamp) => new((long)((Stopwatch.GetTimestamp() - startTimestamp) * TickFrequency));
    
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public static TimeSpan CustomGetElapsedTime2(long startTimestamp) => new((long)((Stopwatch.GetTimestamp() - startTimestamp) * TickFrequency));
    
    public static TimeSpan WatchGetElapsedTime(long startTimestamp) => Stopwatch.GetElapsedTime(startTimestamp);
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public static TimeSpan WatchGetElapsedTime2(long startTimestamp) => Stopwatch.GetElapsedTime(startTimestamp);
}
```

testing result:

![image](https://github.com/user-attachments/assets/7d6fff02-491d-4b64-98fb-2cb980a458b8)

Think maybe we could update the `RateLimiterHelper.GetElapsedTime`

```csharp
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public static TimeSpan GetElapsedTime(long startTimestamp)
#if NET
       => Stopwatch.GetElapsedTime(startTimestamp);
#else
       => new((long)((Stopwatch.GetTimestamp() - startTimestamp) * TickFrequency));
#endif
```

Thoughts?

Reply:
> Think maybe we could update the RateLimiterHelper.GetElapsedTime like following to simplify the call in the RateLimiter

That is essentially the idea I was suggesting.

Notably we also do not care about time differences on the order of 1-5ns for most scenarios. We care much more about the code being understandable, maintainable, etc.

Reply:
updated